src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -85,15 +113,23 @@ case object ThurloeDatabaseConnector extends DataAccess with LazyLogging {
 
   def lookup(userId: String): Future[UserKeyValuePairs] =
     for {
-      results <- lookupWithConstraint(x => x.userId === userId)
+      samUser <- lookupSamUser(userId)
+      results <- lookupWithConstraint(x =>
+        x.userId === samUser.getId || x.userId === samUser.getGoogleSubjectId || x.userId === samUser.getAzureB2CId
+      )
     } yield UserKeyValuePairs(userId, results map { _.userKeyValuePair.keyValuePair })
 
   def lookup(queryParameters: ThurloeQuery): Future[Seq[UserKeyValuePair]] = {
     def userIdAndKeyConstraint(queryParameters: ThurloeQuery) = (x: DbKeyValuePair) => {
       val include: Rep[Boolean] = true
 
       val userIdFilter = queryParameters.userId.map { userIds =>
-        val userIdFilters = userIds map { userId => x.userId === userId }
+        val userIdFilters = userIds map { userId =>
+          val samUser = Await.result(lookupSamUser(userId), 60 seconds)
```

Comment:
This is kind of unfortunate, but I don't really see a better way to fit this lookup in without a large refactor. Since we are removing it eventually and we don't expect high load for this endpoint I think it will be fine.